packages/core/src/types/IResolver.ts

```diff
@@ -59,4 +65,61 @@ export interface IResolver extends IPluginMethodMap {
    * @public
    */
   resolveDid(args: ResolveDidArgs): Promise<DIDResolutionResult>
+
+  /**
+   * Dereference a DID URI fragment and return the corresponding DID document entry.
+   *
+   * @example
+   * ```typescript
+   * const did = 'did:ethr:rinkeby:0xb09b66026ba5909a7cfe99b76875431d2b8d5190'
+   * const didFragment = `${did}#controller`
+   * const fragment = await agent.dereferenceDidUri({
+   *   didDocument: (await agent.resolveDid({didUrl: did}))?.didDocument,
+   *   didURI: didFragment,
+   *   section: 'authentication'
+   * })
+   * expect(fragment).toEqual({
+   *       id: 'did:ethr:rinkeby:0xb09b66026ba5909a7cfe99b76875431d2b8d5190#controller',
+   *       type: 'EcdsaSecp256k1RecoveryMethod2020',
+   *       controller: 'did:ethr:rinkeby:0xb09b66026ba5909a7cfe99b76875431d2b8d5190',
+   *       blockchainAccountId: '0xb09B66026bA5909A7CFE99b76875431D2b8D5190@eip155:4',
+   *     })
+   * ```
+   *
+   * @param args.didDocument - the DID document from which to extract the fragment.
+   *   This MUST be the document resolved by {@link resolveDid}
+   * @param args.didURI - the DID URI that needs to be dereferenced
+   * @param args.section - Optional - the section of the DID Document to be used for dereferencing
+   *
+   * @returns a `Promise` containing the {@link did-resolver#VerificationMethod | VerificationMethod} or
+   *   {@link did-resolver#ServiceEndpoint | ServiceEndpoint}
+   *
+   * @throws `not_found:...` in case the fragment is not displayed in the DID document
+   *
+   * @beta
+   */
+  dereferenceDidUri(args: {
```

Comment:
would it make sense to fully support `dereference` as described in DID Core: https://www.w3.org/TR/did-core/#did-url-dereferencing ? we should probably also add support for that in [did-resolver](https://github.com/decentralized-identity/did-resolver).

Reply:
It would make sense, but the caching mechanism in did-resolver is not yet very well behaved so we might end-up calling resolve() too many times.